apps/projects/app/components/Content/Issues.js

```diff
@@ -559,50 +584,90 @@ class Issues extends React.PureComponent {
                 repoId => downloadedRepos[repoId].hasNextPage
               ).length > 0
 
+            const issuesSorted = this.shapeIssues(issuesFiltered)
+              .sort(currentSorter)
+
+            const estimatedHeight = defaultCardHeight * issuesSorted.length
+
+            const cache = new CellMeasurerCache({
+              defaultHeight: defaultCardHeight,
+              fixedWidth: true
+            })
+
+            const rowRenderer = ({
+              index,
+              isScrolling,
+              isVisible,
+              key,
+              parent,
+              style
+            }) => {
```

Comment:
Unfortunately it does, because it depends on `issuesSorted`, which depends on `issuesFiltered`, which depends on `downloadedIssues`, which depends on `data`, which depends on the `<Query>` component.

Conceptually, this function is in charge of rendering the cell contents, so it depends on the new data being loaded, which in turn depends on re-renders based on the scrolling position.

Reply:
Can we pass `issuesSorted` in as an argument?

Reply:
It [looks like](https://github.com/bvaughn/react-virtualized/blob/master/docs/List.md#rowrenderer) we cannot. Bummer. 

We _could_ do something like this:

```js
const getRowRenderer = ({ issues }) => ({
  index,
  isScrolling,
  isVisible,
  key,
  parent,
  style
}) => { ... }

class Issues extends React.PureComponent {

  ...

  <List rowRenderer={getRowRenderer({ issues: issuesSorted })} ... />
```

I think this would make for more readable code, though I don't think it gets us a performance improvement. What do you think?

Reply:
Wow, that looks just like magic to me. Are you some sort of JavaScript magician? :) I'm gonna try this awesome new trick. Thanks!

Reply:
It worked! I'm also gonna do it for the onScroll event.

Reply:
No, the code ended up messy, and it didn't even work. So I'm just gonna leave the onScroll right were it was.

Reply:
Haha, it might read more clearly in older JS syntax:

```js
function outer(arg1, arg2) {
  return function inner(arg3) {
    // do stuff with all three args
  }
}
```

This pattern is called a [_closure_](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures).